src/chat.rs

```diff
@@ -1873,16 +1898,16 @@ pub(crate) fn add_contact_to_chat_ex(
         msg.param.set_cmd(SystemMessage::MemberAddedToGroup);
         msg.param.set(Param::Arg, contact.get_addr());
         msg.param.set_int(Param::Arg2, from_handshake.into());
+
         msg.id = send_msg(context, chat_id, &mut msg)?;
+    } else {
+        // send an event for unpromoted groups
+        // XXX probably not neccessary because ChatModified should suffice
         context.call_cb(Event::MsgsChanged {
```

Comment:
the XXX comment is wrong here. DC_EVENT_MSGS_CHANGED is needed because there is a new message in the chat and the uis shall redraw the corresponding parts. DC_EVENT_CHAT_MODIFIED only says that chatname/image/memberlist has changed. the uis may react differently to these events.
btw. in the "promoted" if-part, DC_EVENT_MSGS_CHANGED is sent in send_msg()

Reply:
the XXX is at a place where there is *no* message added, therefore the `MsgId==0`. Your argument actually means that the event should not be sent and that ChatModified is precisely the right event for a modified member list without a message ;)

Reply:
you're right. we can remove the whole if-branch.